cluster-autoscaler/cloudprovider/aws/aws_manager.go

```diff
@@ -69,9 +70,10 @@ func CreateAwsManager(configReader io.Reader) (*AwsManager, error) {
 
 	service := autoscaling.New(session.New())
 	manager := &AwsManager{
-		asgs:     make([]*asgInformation, 0),
-		service:  service,
-		asgCache: make(map[AwsRef]*Asg),
+		asgs:                     make([]*asgInformation, 0),
+		service:                  service,
+		asgCache:                 make(map[AwsRef]*Asg),
+		instancesNotInManagedAsg: make(map[AwsRef]struct{}),
```

Comment:
Excuse me I'm not following you correctly but I'd rather suggest extracting a new object dedicated to efficiently fetch ASG data(probably named like `asgRegistry` which has the actual implementation of [`GetAsgForInstance`](https://github.com/kubernetes/autoscaler/pull/46/files/a55e8680777d9fdce4d0a8c78ea6443d9940d789#diff-ade7b95627ea0dd6b6f4deee7f24fa7eR169) if we'd want less complexity here.

After the change, `AwsManager` would just delegate `GetAsgForInstance` to `asgRegistry` and relevant struct members like `asgs`, `asgCache`, `instancesNotInManagedAsg` would move to `asgRegistry`.

The reasoning behind my suggestion is that we won't want to introduce an another gorountine just for calling `regenerateCache` - the less concurrent programming the more deterministic CA's behavior is/the happier our lives are? 😃 

And if you'd just like to `regenerateCache` every X seconds, theoretically, we can just do it by checking if elapsed time since the last regeneration is greater than X, in the very beginning of `regenerateCache`?